src/Plutarch/Test/Export.hs

```diff
@@ -0,0 +1,728 @@
+ provided by the script export is provided as an additional parameter to the
+ function argument.
+
+ Use this similarly to 'modifyingDatum': see its documentation for an example of use.
+
+ @since 1.2.1
+-}
+modifyingScriptParams ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (a :: Type).
+  -- | How to \'locally modify\' the script arguments
+  (info -> ScriptParams dat red -> ScriptParams dat red) ->
+  -- | The collection of tests to run using the modified arguments
+  WithExport dat red info a ->
+  WithExport dat red info a
+modifyingScriptParams f (WithExport comp) = WithExport . local go $ comp
+  where
+    go :: WithExportEnv dat red info -> WithExportEnv dat red info
+    go env =
+      let collected =
+            ScriptParams
+              (view #datum env)
+              (view #redeemer env)
+              (view #context env)
+          modified = f (view #info env) collected
+       in set #datum (view #datum modified)
+            . set #redeemer (view #redeemer modified)
+            . set #context (view #context modified)
+            $ env
+
+{- | As 'modifyingScriptParams', except the \'locally scoped\' arguments are replaced with
+ the given ones.
+
+ @since 1.2.1
+-}
+settingScriptParams ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (a :: Type).
+  -- | How to \'locally set\' the script arguments
+  (info -> ScriptParams dat red) ->
+  -- | The collection of tests to run using the set arguments
+  WithExport dat red info a ->
+  WithExport dat red info a
+settingScriptParams f = modifyingScriptParams (\i _ -> f i)
+
+{- | As 'modifyingScriptParams', but without additional information from the script
+ export.
+
+ @since 1.2.1
+-}
+modifyingScriptParams' ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (a :: Type).
+  -- | How to \'locally modify\' the arguments
+  (ScriptParams dat red -> ScriptParams dat red) ->
+  -- | The collection of tests to run using the modified arguments
+  WithExport dat red info a ->
+  WithExport dat red info a
+modifyingScriptParams' f = modifyingScriptParams (\_ x -> f x)
+
+{- | As 'settingScriptParams', but without additional information from the script
+ export.
+
+ @since 1.2.1
+-}
+settingScriptParams' ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (a :: Type).
+  -- | A \'local value\' for the arguments
+  ScriptParams dat red ->
+  -- | The collection of tests to run using the set arguments
+  WithExport dat red info a ->
+  WithExport dat red info a
+settingScriptParams' x = modifyingScriptParams (\_ _ -> x)
+
+-- Helpers
+
+runWithExport ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (param :: Type).
+  WithExport dat red info () ->
+  RawScriptExport ->
+  Linker param (ScriptExport info) ->
+  TestParams dat red param ->
+  Either LinkerError (Map Text PreparedScriptTests)
+runWithExport (WithExport comp) raw linker args =
+  case runLinker linker raw . view #linkerParams $ args of
```

Comment:
we can technically remove case here and use do notation.

Reply:
Good catch - will do that.